nightshift/constants.py

```diff
@@ -1,86 +1,146 @@
+        "dst_tags2delete": PRINT_TAGS_TO_DELETE,
         "query_days": PRINT_QUERY_DAYS,
     },
     "print_eng_adult_mystery": {
         "nid": 7,
         "description": "Print English adult mysteries",
-        "src_tags2keep": [],
-        "dst_tags2delete": PRINT_TAGS_TO_KEEP,
+        "sierra_format_code": {"NYP": "a", "BPL": "a"},
+        "src_tags2keep": ["910"],
+        "dst_tags2delete": PRINT_TAGS_TO_DELETE,
         "query_days": PRINT_QUERY_DAYS,
     },
     "print_eng_adult_scifi": {
         "nid": 8,
         "description": "Print English adult science-fiction",
-        "src_tags2keep": [],
-        "dst_tags2delete": PRINT_TAGS_TO_KEEP,
+        "sierra_format_code": {"NYP": "a", "BPL": "a"},
+        "src_tags2keep": ["910"],
+        "dst_tags2delete": PRINT_TAGS_TO_DELETE,
         "query_days": PRINT_QUERY_DAYS,
     },
     "print_eng_juv_fic": {
         "nid": 9,
         "description": "Print English juvenile fiction",
-        "src_tags2keep": [],
-        "dst_tags2delete": PRINT_TAGS_TO_KEEP,
+        "sierra_format_code": {"NYP": "a", "BPL": "a"},
+        "src_tags2keep": ["910"],
+        "dst_tags2delete": PRINT_TAGS_TO_DELETE,
         "query_days": PRINT_QUERY_DAYS,
     },
     "print_eng_juv_bio": {
         "nid": 10,
         "description": "Print English juvenile bio",
-        "src_tags2keep": [],
-        "dst_tags2delete": PRINT_TAGS_TO_KEEP,
+        "sierra_format_code": {"NYP": "a", "BPL": "a"},
+        "src_tags2keep": ["910"],
+        "dst_tags2delete": PRINT_TAGS_TO_DELETE,
         "query_days": PRINT_QUERY_DAYS,
     },
     "print_eng_juv_nonfic": {
         "nid": 11,
         "description": "Print English juvenile fiction",
-        "src_tags2keep": [],
-        "dst_tags2delete": PRINT_TAGS_TO_KEEP,
+        "sierra_format_code": {"NYP": "a", "BPL": "a"},
+        "src_tags2keep": ["910"],
+        "dst_tags2delete": PRINT_TAGS_TO_DELETE,
         "query_days": PRINT_QUERY_DAYS,
     },
 }
+
+
+SIERRA_FORMATS = {
+    "NYP": {
+        "ebook": None,
+        "eaudio": None,
+        "evideo": None,
+        "print": "a",
+        "large_print": "l",
+    },
+    "BPL": {
+        "ebook": "x",
+        "eaudio": None,
+        "evideo": None,
+        "print": "a",
+        "large_print": "l",
+    },
+}
+
+
+def library_by_id() -> dict[int, str]:
+    """
+    Creates a dictionary where the key is `datastore.Library.nid` and a value is the
+    library code.
+    """
+    return {v["nid"]: k for k, v in LIBRARIES.items()}
+
+
+def resource_category_by_id() -> dict[int, str]:
+    """
+    Creates a dictionary of resource categories by their id
+    """
+    return {v["nid"]: k for k, v in RESOURCE_CATEGORIES.items()}
+
+
+def sierra_format_code() -> dict[int, dict[str, str]]:
+    """
+    Returns dictionary of sierra codes for each resource category with keys being
+    resource IDs.
+    """
+    return {v["nid"]: v["sierra_format_code"] for v in RESOURCE_CATEGORIES.values()}
```

Comment:
I think this is a very good question. I think it will be better to move this to more general area, especially because I may not be able to address it right away.
Please see issue #12